etcdmain/etcd.go

```diff
@@ -267,13 +267,22 @@ func startEtcd(cfg *config) (<-chan struct{}, error) {
 		plog.Warningf("ignoring client auto TLS since certs given")
 	}
 
-	var ctlscfg *tls.Config
+	var (
+		ctlscfg         *tls.Config
+		ctlscfgInsecure *tls.Config
+	)
 	if !cfg.clientTLSInfo.Empty() {
 		plog.Infof("clientTLS: %s", cfg.clientTLSInfo)
 		ctlscfg, err = cfg.clientTLSInfo.ServerConfig()
 		if err != nil {
 			return nil, err
 		}
+
+		ctlscfgInsecure, err = cfg.clientTLSInfo.ServerConfig()
```

Comment:
@xiang90 Yeah it's a big struct https://golang.org/pkg/crypto/tls/#Config. Seems easier just to create another one?